tools/src/main/java/org/apache/kafka/tools/ConsoleShareConsumerOptions.java

```diff
@@ -0,0 +1,486 @@
+                            output.print(header.key() + ":");
+                            output.write(deserialize(consumerRecord, headersDeserializer, header.value(), consumerRecord.topic()));
+                            if (headersIt.hasNext()) {
+                                output.write(headersSeparator);
+                            }
+                        }
+                    }
+                    writeSeparator(output, printKey || printValue);
+                }
+
+                if (printKey) {
+                    output.write(deserialize(consumerRecord, keyDeserializer, consumerRecord.key(), consumerRecord.topic()));
+                    writeSeparator(output, printValue);
+                }
+
+                if (printValue) {
+                    output.write(deserialize(consumerRecord, valueDeserializer, consumerRecord.value(), consumerRecord.topic()));
+                    output.write(lineSeparator);
+                }
+            } catch (IOException ioe) {
+                LOG.error("Unable to write the consumer record to the output", ioe);
+            }
+        }
+
+        private Map<String, ?> propertiesWithKeyPrefixStripped(String prefix, Map<String, ?> configs) {
+            final Map<String, Object> newConfigs = new HashMap<>();
+            for (Map.Entry<String, ?> entry : configs.entrySet()) {
+                if (entry.getKey().startsWith(prefix) && entry.getKey().length() > prefix.length()) {
+                    newConfigs.put(entry.getKey().substring(prefix.length()), entry.getValue());
+                }
+            }
+            return newConfigs;
+        }
+
+        private byte[] utfBytes(String str) {
+            return str.getBytes(StandardCharsets.UTF_8);
+        }
+
+        private byte[] getByteProperty(Map<String, ?> configs, String key) {
+            return utfBytes((String) configs.get(key));
+        }
+
+        private boolean getBoolProperty(Map<String, ?> configs, String key) {
+            return ((String) configs.get(key)).trim().equalsIgnoreCase("true");
+        }
+
+        private Optional<Deserializer<?>> getDeserializerProperty(Map<String, ?> configs, boolean isKey, String key) {
+            if (configs.containsKey(key)) {
+                String name = (String) configs.get(key);
+                try {
+                    Deserializer<?> deserializer = (Deserializer<?>) Class.forName(name).getDeclaredConstructor().newInstance();
+                    Map<String, ?> deserializerConfig = propertiesWithKeyPrefixStripped(key + ".", configs);
+                    deserializer.configure(deserializerConfig, isKey);
+                    return Optional.of(deserializer);
+                } catch (Exception e) {
+                    LOG.error("Unable to instantiate a deserializer from " + name, e);
+                }
+            }
+            return Optional.empty();
+        }
+    }
+
+    public static class LoggingMessageFormatter implements MessageFormatter {
```

Comment:
This formatter and the next one could probably be omitted in this PR. I understand that they will appear when the console consumer PR is merged and that's fine.